### ISSUE TITLE
[Bug]: Memory Leak

### ISSUE THREAD

**Issue by Althea44** [TeamGalacticraft/Galacticraft-Legacy#182]
### Forge Version

14.23.5.2859

### Galacticraft Version

4.0.6

### Reproduction steps

Existence of any Space Suit on players or armor stands causes a steady memory leak that leads to my game crashing. As long as I am in the render distance of the said Space Suits, whether I look at them or not causes this. I am playing on a server with 4 other people and they don't seem to be effected by this issue, because of this I redownloaded my java and Minecraft but this doesn't seem to fix the issue.

**Comment by Althea44** [comment]
I will add the list of mods and make the game crash once again to get its report as I have lost all the previous ones after deleting and redownloading the game.

**Comment by Althea44** [comment]
I joined a super-flat world and spawned in nothing but Space Suits, put them on armor stands and it increased my memory usage around 1gb per second until it crashed with the code "-1073741819". It didn't create a crash report but I did take a photo of my mod list. I don't know what else I need to provide besides a crash report which I will try to get by making my game crash more.
![mod list](https://github.com/TeamGalacticraft/Galacticraft-Legacy/assets/164206463/d8842ddc-9b93-497c-bcfd-6dc2d5c606af)
